src/groupByKey.ts

```diff
@@ -2,6 +2,17 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const grouppedItems: GroupsMap<T> = {};
+
+  items.forEach((item) => {
+    const itemKey = item[key] as string;
```

Comment:
let's not convert types unless we really need to. You can convert the value instead